packages/core/src/components/Table/context/TableContext/TableContext.tsx

```diff
@@ -1,17 +1,63 @@
-import React, { createContext, useContext, useMemo, useState } from "react";
+import React, { createContext, UIEventHandler, useCallback, useContext, useMemo, useRef, useState } from "react";
 import { ITableContext, ITableProviderProps } from "./TableContext.types";
 
 const TableContext = createContext<ITableContext | undefined>(undefined);
 
 export const TableProvider = ({ value, children }: ITableProviderProps) => {
-  const [rowWidth, setRowWidth] = useState<number>(0);
-  const contextValue = useMemo(
+  const [isVirtualized, setIsVirtualized] = useState<boolean>(false);
+  const markTableAsVirtualized = useCallback(() => {
+    setIsVirtualized(true);
+  }, []);
+
+  const [scrollLeft, setScrollLeft] = useState<number>(0);
+  const headRef = useRef<HTMLDivElement>(null);
```

Comment:
What's this? What's the usage of the head here?

Reply:
to scroll between the header and body horizontal scrolls, as both are always visually appear but behind the container's limits